sdk/src/esplora.ts

```diff
@@ -388,12 +389,19 @@ export class EsploraClient {
     }
 
     /**
-     * Get the balance for an account / address.
+     * Get the Bitcoin balance for an address, returning both confirmed (on-chain) and unconfirmed (mempool) balances.
+     *
+     * @dev This method returns an object containing the confirmed balance, unconfirmed balance, and the total balance.
+     *      The confirmed balance represents the amount that is on-chain, while the mempool balance includes transactions
+     *      that are pending (unconfirmed). The total is the sum of both confirmed and unconfirmed balances.
      *
      * @param {string} address - The Bitcoin address to check.
-     * @returns {Promise<number>} A promise that resolves to the balance.
+     * @returns {Promise<{ chain: number, mempool: number, total: number }>} A promise that resolves to an object containing:
+     *      - `chain`: The confirmed on-chain balance in satoshis.
+     *      - `mempool`: The unconfirmed balance (in mempool) in satoshis.
+     *      - `total`: The total balance, which is the sum of the confirmed and unconfirmed balances.
```

Comment:
I would've preferred `confirmed`, `unconfirmed/pending` and `total` but approving anyway because it doesn't matter :)